pycalphad/core/custom_autowrap.py

```diff
@@ -128,58 +127,274 @@ def import_extension(path, modname):
 class CodeWrapError(Exception):
     pass
 
+class C89CodePrinter(CCodePrinter):
+    def _get_loop_opening_ending(self, indices):
+        # The purpose is to enable C89-compliant loops (indices are pre-declared)
+        open_lines = []
+        close_lines = []
+        loopstart = "for (%(var)s=%(start)s; %(var)s<%(end)s; %(var)s++){"
+        for i in indices:
+            # C arrays start at 0 and end at dimension-1
+            open_lines.append(loopstart % {
+                'var': self._print(i.label),
+                'start': self._print(i.lower),
+                'end': self._print(i.upper + 1)})
+            close_lines.append("}")
+        return open_lines, close_lines
+
+class CustomCCodeGen(CCodeGen):
+    def get_prototype(self, routine):
+        """Returns a string for the function prototype of the routine.
```

Comment:
I might change the word `prototype` in the docstring to `declaration` and remove the wikipedia link, but this is fine.